bundle.go

```diff
@@ -0,0 +1,151 @@
+	"path/filepath"
+	"strings"
+
+	spec "github.com/opencontainers/specs/specs-go"
+)
+
+const image = "busybox"
+
+const tmpDir = "/tmp"
+
+const configTemplate = "src/github.com/clearcontainers/tests/data/config.json"
+
+// Bundle represents the root directory where config.json and rootfs are
+type Bundle struct {
+	// Config represents the config.json
+	Config *spec.Spec
+
+	// Path to the bundle
+	Path string
+}
+
+// NewBundle creates a new bundle
+func NewBundle(workload []string) (*Bundle, error) {
+	path, err := ioutil.TempDir(tmpDir, "bundle")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := createRootfs(path); err != nil {
+		return nil, err
+	}
+
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		return nil, fmt.Errorf("GOPATH is not set")
+	}
+
+	configTemplatePath := filepath.Join(gopath, configTemplate)
+	content, err := ioutil.ReadFile(configTemplatePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config.json template: %s", err)
+	}
+
+	var config spec.Spec
+	err = json.Unmarshal(content, &config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config.json template: %s", err)
+	}
+
+	config.Process.Args = workload
+
+	bundle := &Bundle{
+		Path:   path,
+		Config: &config,
+	}
+
+	err = bundle.Save()
+	if err != nil {
+		return nil, err
+	}
+
+	return bundle, nil
+}
+
+// createRootfs creates a rootfs in the specific bundlePath
+func createRootfs(bundlePath string) error {
+	rootfsDir := filepath.Join(bundlePath, "rootfs")
+	if err := os.MkdirAll(rootfsDir, 0755); err != nil {
+		return err
+	}
+
+	// create container
+	var container bytes.Buffer
+	createCmd := exec.Command("docker", "create", image)
+	createCmd.Stdout = &container
+	if err := createCmd.Run(); err != nil {
+		return fmt.Errorf("Failed to create container: %s", err)
+	}
+	containerName := strings.TrimRight(container.String(), "\n")
+
+	// export container
+	tarFile, err := ioutil.TempFile(tmpDir, "tar")
+	if err != nil {
+		return fmt.Errorf("Failed to create temporal file: %s", err)
+	}
+	defer tarFile.Close()
+
+	exportCmd := exec.Command("docker", "export", "-o", tarFile.Name(), containerName)
+	if err := exportCmd.Run(); err != nil {
+		return fmt.Errorf("Failed to export container: %s", err)
+	}
+	defer os.Remove(tarFile.Name())
+
+	// extract container
+	tarCmd := exec.Command("tar", "-C", rootfsDir, "-pxf", tarFile.Name())
+	if err := tarCmd.Run(); err != nil {
+		return fmt.Errorf("Failed to extract container: %s", err)
+	}
+
+	// remove container
+	rmCmd := exec.Command("docker", "rm", "-f", containerName)
+	if err := rmCmd.Run(); err != nil {
+		return fmt.Errorf("Failed to remove container: %s", err)
+	}
+
+	return nil
+}
+
+// Save to disk the Config
+func (b *Bundle) Save() error {
+	content, err := json.Marshal(b.Config)
+	if err != nil {
+		return fmt.Errorf("failed to marshal config: %s", err)
+	}
+
+	configFile := filepath.Join(b.Path, "config.json")
+	err = ioutil.WriteFile(configFile, content, 0664)
```

Comment:
yes ```0644``` is ok

> What user do we expect this code to run as?

I think only root is able to start containers

> Do we need to assert that?

I do not think so